snap/gui/zed.desktop

```diff
@@ -0,0 +1,17 @@
+[Desktop Entry]
```

Comment:
I believe you don't need top copy the desktop file. See my comment on the snapcraft forum:  https://forum.snapcraft.io/t/zed-rust-based-ide/40993/8

The main idea is to do this:
```
    override-stage: |
      sed -i -e 's,Icon=zed,Icon=${SNAP}/share/icons/hicolor/1024x1024/apps/zed.png,' "$CRAFT_PART_INSTALL"/share/applications/zed.desktop
      craftctl default
```

Reply:
The problem is when using rust plugin, the desktop file will not be in the final set files. Also, the file has placeholders. This gets rid of all those complications.

Reply:
I don't think we should ship multiple copies of the same stuff in the repo. I think right way to do this is to wrap the snap invocation in a shell-script that produces the input files it expects from the same files we use to build the other packages.

Reply:
Can you kindly share some example with us? What do you mean by wrap the snap invocation in a shell-script?

Reply:
https://github.com/zed-industries/zed/blob/main/script/flatpak/bundle-flatpak is how this is done for flatpak for example.

Reply:
This will be run by a CI?